File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        list<char> myList;
        int left = 0;
        int maxlen = 0;
        for (int right = 0; right < s.length(); right++) {
            if (find(myList.begin(), myList.end(), s[right]) == myList.end()) {
                myList.push_back(s[right]);
                maxlen = max(maxlen, (int)myList.size());
            } else {
                myList.remove(s[left]);
                left++;
                right--; // Decrement right to recheck the current character
            }
        }
        return maxlen;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> h=new HashSet<>();
        int left=0;
        int max=0;
        for(int right=0;right<s.length();right++){
            while(h.contains(s.charAt(right))){
                h.remove(s.charAt(left));
                left++;
            }
            h.add(s.charAt(right));
            max=Math.max(max,right-left+1);
        } 
        return max;
    }
}